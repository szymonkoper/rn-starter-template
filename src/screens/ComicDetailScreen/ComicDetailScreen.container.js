import React from 'react'
import navigationShape from 'navigation/proptypes'
import ComicDetailScreen from './ComicDetailScreen'

const ComicDetailScreenContainer = props => {
  const { navigation } = props
  const imgSource = navigation.getParam('imgSource', null)

  // eslint-disable-next-line react/jsx-props-no-spreading
  return <ComicDetailScreen imgSource={imgSource} {...props} />
}

ComicDetailScreenContainer.propTypes = {
  navigation: navigationShape.isRequired
}

export default ComicDetailScreenContainer
