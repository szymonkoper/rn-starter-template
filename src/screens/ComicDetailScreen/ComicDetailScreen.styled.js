import styled from 'styled-components/native'
import { backgroundColorPrimary } from 'style/colors'

export const ScreenContainer = styled.SafeAreaView`
  background-color: ${backgroundColorPrimary};
  flex: 1;
`

export const ComicImage = styled.Image.attrs(() => ({ resizeMode: 'contain' }))`
  flex: 1;
`
