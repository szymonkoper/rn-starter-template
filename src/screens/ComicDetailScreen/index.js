export { default } from './ComicDetailScreen.container'
