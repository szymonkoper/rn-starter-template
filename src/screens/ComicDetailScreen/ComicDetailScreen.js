import React from 'react'
import PropTypes from 'prop-types'
import { ComicImage, ScreenContainer } from './ComicDetailScreen.styled'

const ComicDetailScreen = ({ imgSource }) => (
  <ScreenContainer>
    <ComicImage source={{ uri: imgSource }} />
  </ScreenContainer>
)

ComicDetailScreen.propTypes = {
  imgSource: PropTypes.string.isRequired
}

export default ComicDetailScreen
