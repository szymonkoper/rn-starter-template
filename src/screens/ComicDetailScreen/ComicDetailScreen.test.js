import React from 'react'
import { render } from 'react-native-testing-library'
import ComicDetailScreen from './ComicDetailScreen'

describe('ComicDetailScreen', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <ComicDetailScreen imgSource="http://example.com/image.jpg" />
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
