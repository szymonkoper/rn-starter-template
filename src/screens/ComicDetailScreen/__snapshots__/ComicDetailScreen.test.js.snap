// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicDetailScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "http://example.com/image.jpg",
      }
    }
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
</RCTSafeAreaView>
`;
